### PR TITLE
Remove format option & update setting.dest with option of `'none'`

### PR DESCRIPTION
**Pull request by tisAliG**
#32 & #34

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Sugarcoat #
+# Sugarcoat #
```
Comment (vbalcom) [inline README.md]: Oops a trailing space!

File: generators/pattern-library/configure.js [hunk 2]
```diff
@@ -111,19 +110,69 @@ function init( options ) {
...
+
+    }
+    else if ( settings.dest ) {
+
         settings.dest = path.resolve( settings.cwd, settings.dest );
+
+    }
+    else {
+
+        error = error || [];
```
Comment (vbalcom) [inline generators/pattern-library/configure.js]: Make error an array from the very beginning, and push errors as configure runs. Then we check for the length of the error array to determine if there are errors in our test.

File: generators/pattern-library/configure.js [hunk 3]
```diff
@@ -111,19 +110,69 @@ function init( options ) {
...
+        error.push({
+            'key': 'settings.dest',
```
Comment (vbalcom) [inline generators/pattern-library/configure.js]: `key` is a valid identifier, no need for quotes. same with within the rest of this file

File: generators/pattern-library/index.js [hunk 4]
```diff
@@ -16,22 +16,23 @@ function init( config ) {
...
-        log.error(config.error);
-
-        return Promise.reject();
+        return new Promise( function ( resolve, reject ) {
+            return reject( config );
+        });
     }
 
     return globFiles( config )
+    .then( globFiles )
```
Comment (vbalcom) [inline generators/pattern-library/index.js]: Why is globFiles happening twice?

Comment (tisAliG) [inline generators/pattern-library/index.js]: whoops, must have forgot to remove that when I was making changes.

File: generators/pattern-library/index.js [hunk 5]
```diff
@@ -16,22 +16,23 @@ function init( config ) {
...
     .then( readSections )
     .then( parseSections )
     .then( render )
     .then( function ( html ) {
 
         log.info( 'Finished!' );
 
-        return output( html, config );
+        return html;
```
Comment (vbalcom) [inline generators/pattern-library/index.js]: If we take a look at output(), the function you've removed here, its returning the config file when format isn't declared. In our docs, we've also stated that we should be returning the expanded config object. So lets make sure we're returning the config object in this last return statement.

File: test/configure.js [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+                },
+                {
+                    title: 'CSS File 2',
+                    files: './test/assert/parseVarCode.css'
+                }
+            ]
+        };
+
+        sugarcoat( configMissingDest )
+        .then( data => {
```
Comment (vbalcom) [inline test/configure.js]: Need to assert in the resolve function as well as the reject. Say if Sugarcoat fails to reject when a dest is not provided, you need an assertion to say that this failed. It can be something simple!

File: test/configure.js [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+var fs = require( 'fs-extra' );
+var path = require( 'path' );
+
+var sugarcoat = require( '../index' );
+
+suite( 'Configure: Settings', () => {
+
+    test( 'Destination is set to be required. Destination errored out when not supplied.', (done) => {
+
+        var configMissingDest = {
```
Comment (vbalcom) [inline test/configure.js]: Since we're testing here a completely empty "settings" object, what do you think about testing for all the errors that are returned?

File: test/configure.js [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+
+                }
+            ]
+        };
+
+        sugarcoat( configNoDest )
+        .then( data => {
+
+            assert.isNull( data.settings.dest, 'The destination was set to null' );
+            assert.isFalse( fs.existsSync( path.relative( data.settings.cwd, 'index.html' ) ), 'Sugarcoat did not create a index.html file.' );
```
Comment (vbalcom) [inline test/configure.js]: Since we're going to have a lot of tests running, we should be running all async tests

File: test/configure.js [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+
+            done();
+        });
+
+    });
+});
+
+suite( 'Configure: Sections', () => {
+
+    test( 'Title is set to be required. Title errored out when title is the only required option that was not supplied.', done => {
```
Comment (vbalcom) [inline test/configure.js]: Isn't title optional?

Comment (vbalcom) [inline test/configure.js]: I think I see my confusion. You've added `settings.title` so I assumed you were testing that. Let's remove line 84

File: test/configure.js [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+            if ( data.length === 1 ) {
+
+                assert.propertyVal( data[0], 'key', 'sections.title', 'We have sections.title in the error array.' );
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out when title and dest were not supplied.', done => {
```
Comment (vbalcom) [inline test/configure.js]: We've already tested when dest is not supplied. For unit tests, we usually want to cover one change at a time, and not an accumulation of small changes.

File: test/configure.js (hunk 7 above)
Comment (vbalcom) [inline test/configure.js]: Actually, I realize we should be only testing one small change at a time, so you can ignore this.

File: test/configure.js [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
+                    break;
+                }
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out when all required options were not supplied.', done => {
```
Comment (vbalcom) [inline test/configure.js]: This test, to me, looks more like you're testing if the sections array is empty. Let's revise the description

File: test/configure.js [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+            sections: []
+        };
+
+        sugarcoat( configMissingTitleFiles )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
```
Comment (vbalcom) [inline test/configure.js]: Let's make this statement more specific to the test, since it can output multiple times

File: test/configure.js [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.files' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
```
Comment (vbalcom) [inline test/configure.js]: Now that we have two different sections that are missing titles, the errors are both identical. Is this helpful for debugging? Should we be providing the error's files value as well?

Comment (tisAliG) [inline test/configure.js]: We are returning the entire section object in the error that gets logged. Should we be testing to make sure they are different?

File: test/configure.js [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+            done();
+        });
+    });
+
+    test( 'Files is set to be required. Files errored out when all required options were not supplied.', done => {
+
+        var configMissingTitleFiles = {
+            sections: []
+        };
+
```
Comment (vbalcom) [inline test/configure.js]: Isn't this config test the same as line 145?

File: test/configure.js (hunk 13 above)
Comment (vbalcom) [inline test/configure.js]: I think you'd have to test not the key (they aren't unique), but the section object that is being returned. Like title = 'CSS File 2'

**Comment by tisAliG** [comment]
Your comment that says we need more tests. `2. When the settings array does not exist`, do you mean sections array?

File: test/configure.js [hunk 15]
```diff
@@ -93,6 +103,7 @@ suite( 'Configure: Sections', () => {
         sugarcoat( configMissingOnlyTitle )
         .then( data => {
 
+            assert.isArray( data, 'Sugarcoat should be erroring out.' );
```
Comment (vbalcom) [inline test/configure.js]: Let's reference the test in the assertion sentence here as well

File: test/configure.js [hunk 16]
```diff
@@ -140,7 +152,7 @@ suite( 'Configure: Sections', () => {
         });
     });
 
-    test( 'Title is set to be required. Title errored out when all required options were not supplied.', done => {
+    test.only( 'Title is set to be required. Title errored out when all required options were not supplied.', done => {
```
Comment (vbalcom) [inline test/configure.js]: Before merge make sure to get rid of the `.only`

File: test/configure.js (hunk 13 above)
Comment (tisAliG) [inline test/configure.js]: I'm a little confused as to what we would be testing here? When sugarcoat errors out, we don't get the config object back, we get the error obj which in this case the message includes the object that has the error within the errror message like so: `Files is required. Please add a 'files' option to section object: { title: 'CSS File' }`. The only way to check if the title is not being used twice would be if we did a `.notMatch` which we would need to create a regex expression to make sure that the title is not being used in more than one error. Which seems overly complicated for this test. Is that what you mean? Or do you see another way of doing this?

Comment (ryanfitzer) [inline test/configure.js]: > When sugarcoat errors out, we don't get the config object back, we get the error obj

I would expect that we get back the config object with an `errors` property. Is that not that the case?

Comment (vbalcom) [inline test/configure.js]: @tisAliG I think we discussed this last week - and we agreed that since title is being returned to the error object, we should test the title, as opposed to the error object purely existing

Comment (tisAliG) [inline test/configure.js]: @vbalcom Why would we test the title though? In the test above, we are looking to test the errors for when multiple section.files are missing. We want to make sure that those errors are in the error object. 

@ryanfitzer About a week or two ago, Valerie and I came to the conclusion that when there is an error we want the user to only get those errors. Not the entire config, since they shouldn't be able to do anything with the config object when there are errors present.

Comment (vbalcom) [inline test/configure.js]: @tisAliG We want to test against something that would be distinct enough that when the test fails, its obvious why. It's better to test two unique IDs rather than two objects that are identical.

Comment (ryanfitzer) [inline test/configure.js]: > Valerie and I came to the conclusion that when there is an error we want the user to only get those errors.

How will they know when they're getting an error object vs. the config object?

Comment (tisAliG) [inline test/configure.js]: @ryanfitzer It's an error array, rather than the config object.

Comment (vbalcom) [inline test/configure.js]: > How will they know when they're getting an error object vs. the config object?
The errors are returned in an array instead of an object. Ali checks for an array to see if there was an error

Comment (ryanfitzer) [inline test/configure.js]: I don't think type-checking should be the way a user knows if something went wrong. I see 2 options:

1. Add an `errors` property to the returned object.
2. Call `reject` to resolve the promise.

If fact, I think we should do both. Use the first scenario on non-critical errors. The second for critical ones.

Comment (vbalcom) [inline test/configure.js]: @ryanfitzer That sounds like a good idea. We're already rejecting the promise when something goes wrong. We'll just need to move the errors array to a property in the config object then.

Comment (tisAliG) [inline test/configure.js]: @ryanfitzer and I just had a conversation about this. We came to the conclusion that we should be erroring out and rejecting on the first error that we encounter. That way, we don't have to worry about adding an errors array to the config, and we don't have to worry about gathering up all of the errors before they are logged out. 

We are also going to use the Error constructor to create an error object that will be passed to npm log and the rejected promise.

**Comment by tisAliG** [comment]
reverted error messaging to what it was the other day as it was in an ok place. Error messaging for configure will go in a new branch.
